lib/Doctrine/DBAL/SQLParserUtils.php

```diff
@@ -98,7 +103,7 @@ static public function expandListParameters($query, $params, $types)
         foreach ($types as $name => $type) {
             ++$bindIndex;
 
-            if ($type !== Connection::PARAM_INT_ARRAY && $type !== Connection::PARAM_STR_ARRAY) {
+            if (!in_array($type, $relevantArrayTypes)) {
```

Comment:
I agree that using `in_array()` might be cleaner code but unfortunately this method here is performance critical so it is better to add the third comparison instead.